FreeSurfer/LongitudinalFreeSurferPipeline.sh

```diff
@@ -0,0 +1,368 @@
+#!/bin/bash
+
+#~ND~FORMAT~MARKDOWN~
+#~ND~START~
+#
+# # LongitudinalFreeSurferPipeline.sh
+#
+# ## Copyright Notice
+#
+# Copyright (C) 2015-2023 The Human Connectome Project/Connectome Coordination Facility
+#
+# * Washington University in St. Louis
+# * Univeristy of Ljubljana
+#
+# ## Author(s)
+#
+# * Jure Demsar, Faculty of Computer and Information Science, University of Ljubljana
+# * Matthew F. Glasser, Department of Anatomy and Neurobiology, Washington University in St. Louis
+# * Mikhail V. Milchenko, Department of Radiology, Washington University in St. Louis
+#
+
+# Version: v.0.1, 10/23/2023
+
+# ## Product
+#
+# [Human Connectome Project](http://www.humanconnectome.org) (HCP) Pipelines
+#
+# ## License
+#
+# See the [LICENSE](https://github.com/Washington-University/HCPPipelines/blob/master/LICENSE.md) file
+#
+#~ND~END~
+
+#  Define Sources and pipe-dir
+# -----------------------------------------------------------------------------------
+
+set -eu
+
+pipedirguessed=0
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+    pipedirguessed=1
+    #fix this if the script is more than one level below HCPPIPEDIR
+    export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
+
+source "${HCPPIPEDIR}/global/scripts/debug.shlib" "$@"           # Debugging functions; also sources log.shlib
+source "${HCPPIPEDIR}/global/scripts/newopts.shlib" "$@"
+source "${HCPPIPEDIR}/global/scripts/processingmodecheck.shlib"  # Check processing mode requirements
+source "$HCPPIPEDIR/global/scripts/parallel.shlib" "$@"
+
+#description to use in usage - syntax of parameters is now explained automatically
+opts_SetScriptDescription "Runs the Longitudinal FreeSurfer HCP pipeline"
+
+# Show usage information
+opts_AddMandatory '--subject' 'SubjectID' 'subject' "Subject ID (required)  Used with --path input to create full path to root directory for all sessions"
+opts_AddMandatory '--path' 'StudyFolder' 'path' "Path to subject's data folder (required)  Used with --subject input to create full path to root directory for all outputs generated as path/subject)"
+opts_AddMandatory '--sessions' 'Sessions' 'sessions' "Comma separated list of session (timepoint, visit) IDs (required). Also used to generate full path to each longitudinal session directory"
+opts_AddMandatory '--longitudinal-template' 'TemplateID' 'template-id' "Longitudinal template label"
+opts_AddOptional '--use-T2w' 'UseT2w' 'UseT2w' "Set to 0 for no T2-weighted processing [1]" "1"
```

Comment:
Needs to use `opts_StringToBool` - should this be a "legacy processing" flag?

Reply:
Not sure what you mean here..

Reply:
`--use-T2w` is clearly a boolean-like flag, but it only works as expected for the exact strings "0" and "1", and there is currently no sanity checking.  You should use opts_StringToBool for sanity checking its value, and to extend its accepted values to "yes", "true", etc:

https://github.com/Washington-University/HCPpipelines/blob/9cfa37a378dff26fd5aa4fd2aad24d0136830f0f/global/scripts/RSNregression.sh#L73

As for "legacy processing", in cross-sectional freesurfer, when people run it without a T2w image, we consider it a legacy processing mode (that results in worse cortical segmentation), and require the user to affirm that they know they are not running the pipelines in a recommended configuration for modern datasets.  However, if the user needs to manually run cross-sectional freesurfer (with ordinary PreFS) before they can run anything longitudinal, then that may already have taken care of informing the user about the caveats of not using a T2w image, so maybe we don't have to admonish them again here.

Reply:
Ok. I added opts_StringToBool conversion to UseT2w